src/session_impl.cpp

```diff
@@ -850,6 +835,19 @@ namespace aux {
 		}
 #endif
 
+		settings = e->dict_find_dict("settings");
+		if (settings)
+		{
+			// apply_settings_pack will update dht and proxy
+			boost::shared_ptr<settings_pack> pack = load_pack_from_dict(settings);
+			apply_settings_pack(pack);
+		}
+		else
```

Comment:
since this only applies to builds with deprecated functions enabled, could you wrap this else statement in #ifndef TORRENT_NO_DEPRECATE too?